exercises/simple-linked-list/src/main/scala/SimpleLinkedList.scala

```diff
@@ -0,0 +1,10 @@
+import scala.reflect.ClassTag
+
+trait SimpleLinkedList[T] {
+  def isEmpty: Boolean
+  def value: T
+  def add(item: T): SimpleLinkedList[T]
+  def next: SimpleLinkedList[T]
+  def reverse: SimpleLinkedList[T]
+  def toArray(implicit m: ClassTag[T]): Array[T]
```

Comment:
How do you feel about the README for the exercise specifically taking about Arrays?

From the README - 
_As a first take, lets create a singly linked list to contain the range (1..10), and provide functions to reverse a linked list and convert to and from arrays._